apps/designer/app/designer/features/style-panel/sections/flex-child/flex-child.tsx

```diff
@@ -0,0 +1,267 @@
+              label: "Grow if possible",
+              value: "grow",
+            },
+            {
+              child: <FlexSizingShrinkIcon />,
+              label: "Shrink if needed",
+              value: "shrink",
+            },
+            {
+              child: (
+                <ValuePickerPopover
+                  title="Sizing"
+                  content={
+                    <Grid
+                      css={{
+                        gridTemplateColumns: "1.5fr 1fr 1fr",
+                        gap: "$spacing$9",
+                        padding: "$spacing$9",
+                      }}
+                    >
+                      <Grid
+                        css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}
+                      >
+                        <PropertyName
+                          property={
+                            sectionStyle.flexBasis?.styleConfig.property
+                          }
+                          label="Basis"
+                        />
+                        <TextControl {...sectionStyle.flexBasis} />
+                      </Grid>
+                      <Grid
+                        css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}
+                      >
+                        <PropertyName
+                          property={sectionStyle.flexGrow?.styleConfig.property}
+                          label="Grow"
+                        />
+                        <TextControl {...sectionStyle.flexGrow} />
+                      </Grid>
+                      <Grid
+                        css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}
+                      >
+                        <PropertyName
+                          property={
+                            sectionStyle.flexShrink?.styleConfig.property
+                          }
+                          label="Shrink"
+                        />
+                        <TextControl {...sectionStyle.flexShrink} />
+                      </Grid>
+                    </Grid>
+                  }
+                >
+                  <Flex>
+                    <EllipsesIcon />
+                  </Flex>
+                </ValuePickerPopover>
+              ),
+              label: "More sizing options",
+              value: "",
+            },
+          ]}
+        />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "4fr auto" }}>
+        <PropertyName
+          property={sectionStyle.order?.styleConfig.property}
+          label="Order"
+        />
+        <ToggleGroupControl
+          property={sectionStyle.order?.styleConfig.property}
+          onValueChange={(value) => {
+            switch (value) {
+              case "0":
+              case "1":
+              case "-1": {
+                setOrder(value);
+                break;
+              }
+            }
+          }}
+          value={String(currentStyle.order?.value)}
+          items={[
```

Comment:
same thing here, would it make sense for it to be separate component? (same file)

Reply:
rather a function